3668 intra case correlation content viewer

aziehl opened the pull request:
Pull for VIK-3688 completion.

APriestman commented:
Please resolve the merge conflicts

APriestman commented on Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java:
```diff
@@ -414,64 +422,130 @@ private AbstractFile getAbstractFileFromNode(Node node) {
+            for (ContentTag tag : fileMatchTags) {
+                TskData.FileKnown tagKnownStatus = tag.getName().getKnownStatus();
+                if (tagKnownStatus.compareTo(TskData.FileKnown.KNOWN) == 0 || tagKnownStatus.compareTo(TskData.FileKnown.BAD) == 0) {
+                    caseDbInstance.setKnownStatus(TskData.FileKnown.BAD);
```
Is this saying that if the tag status is either KNOWN or BAD, then set the known status to BAD? That doesn't seem right. I don't think tags can have the KNOWN status anyway so we can probably just test for BAD.

APriestman commented on Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java:
```diff
@@ -414,64 +422,130 @@ private AbstractFile getAbstractFileFromNode(Node node) {
+        }
+
+        // If known, or not in CR, add
+        if (knownStatus.compareTo(TskData.FileKnown.KNOWN) == 0 || knownStatus.compareTo(TskData.FileKnown.BAD) == 0 || !artifactInstances.containsKey(instKey)) {
```
I feel like using "equals" instead of "compareTo" would make this code (and other similar code) more readable. This would change to: 
if (knownStatus.equals(TskData.FileKnown.KNOWN) || knownStatus.equals(TskData.FileKnown.BAD) || ...

aziehl commented:
Ann, I resolved the merge conflicts and incorporated your feedback. I removed the Known checks since you were right, the functionality appears to be the same just checking equality on Bad.

APriestman commented on Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/ArtifactKey.java:
```diff
@@ -0,0 +1,70 @@
+ */
+final class ArtifactKey {
+    
+    private final String dataSource;
```
I think this would be more clear if you renamed this to dataSourceDeviceID (or similar) and updated  the name of getDataSource(). The constructor is a bit ambiguous as it stands.

APriestman commented on Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java:
```diff
@@ -414,64 +422,131 @@ private AbstractFile getAbstractFileFromNode(Node node) {
+        }
+
+        // If known, or not in CR, add
+        if (knownStatus.equals(knownBad) || !artifactInstances.containsKey(instKey)) {
```
I think you need to get the known status from caseDbInstance here instead of using the original value, since it may have been updated if a bad tag was found?

APriestman commented:
Looks good - you should just need to fix the new conflict that popped up.

APriestman commented on Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java:
```diff
@@ -414,64 +422,133 @@ private AbstractFile getAbstractFileFromNode(Node node) {
+        } else {
+            try {
+                // If EamDb not enabled, get the Files default correlation type to allow Other Occurances to be enabled.   
+                if(this.file != null) {
```
Sorry one more thing. I think you need to do the same checks from above about the MD5 being null or empty. I'm getting stack traces in the logs when I click on things that have no hash (for example, an unalloc space file)

aziehl commented:
Thanks, I'll make that fix, resolve the conflict and bring this up to the latest develop.

aziehl commented:
I've added the md5 null check, updated the code to pass codeacy and merged in the latest develop. I changed getOpenCase() to use getCurrentCase(), hopefully that's the correct usage versus getCurrenCaseThrows().
